### PR TITLE
Fix multi value column index creation

### PR DESCRIPTION
**Pull request by saurabhd336**
Fixes https://github.com/apache/pinot/issues/8847

### CODE REVIEW

**Comment by KKcorps** [comment]
Any tests to check for this scenario

**Comment by saurabhd336** [comment]
> Any tests to check for this scenario

Yes I'm adding tests to this PR

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java [hunk 1]
```diff
@@ -386,30 +386,78 @@ public void indexRow(GenericRow row)
...
+          for (Object value: values) {
+            bloomFilterCreator.add(value.toString());
+          }
+        }
       }
 
       // range index
       CombinedInvertedIndexCreator combinedInvertedIndexCreator = _rangeIndexFilterCreatorMap.get(columnName);
       if (combinedInvertedIndexCreator != null) {
         if (dictionaryCreator != null) {
```
Comment (Jackie-Jiang) [inline pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java:402]: Suggest for checking SV then dictionary. MV is always dictionary encoded:
```
if (fieldSpec.isSingleValueField()) {
  if (dictionaryCreator != null) {
    ...
  } else {
    switch ...
  }
} else {
  assert dictionaryCreator != null;
  ...
}

**Review by Jackie-Jiang — APPROVED** [review]
LGTM. Let's add some tests to prevent regression

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java (hunk 1 above)
Comment (saurabhd336) [reply pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java:402]: I was following the segment loading part, and there seems to be handling for both single and multi value for non dictionary encoded columns.. Should be adding this constraint in the offline path?

https://github.com/apache/pinot/blob/master/pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/invertedindex/RangeIndexHandler.java#L170

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/OfflineSegmentCreationMultiValueIndexingTest.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    List<File> jsonFiles = unpackAvroData(_tempDir);
+    ClusterIntegrationTestUtils.buildSegmentFromJson(jsonFiles.get(0), tableConfig, schema, 0, _segmentDir, _tarDir);
+    uploadSegments(getTableName(), TableType.OFFLINE, _tarDir);
+    waitForAllDocsLoaded(600_000L);
+  }
+
+  @Test
+  public void testMvBloomFilterColumnCount()
+      throws Exception {
+    JsonNode queryResponse = postQuery("SELECT COUNT(*) FROM OFFLINE_MULTIVALUE_TABLE WHERE interests = 'i1'");
```
Comment (npawar) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/OfflineSegmentCreationMultiValueIndexingTest.java:92]: mind adding some more queries like IN, valueIn, != ?

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java:402]: Good point. Let's keep them consistent

**Pull request merged**